### PR TITLE
HPCC-14997 Error 'First parameter of FETCH should be a disk file'

### PR DESCRIPTION
**Pull request by shamser**
Signed-off-by: Shamser Ahmed shamser.ahmed@lexisnexis.co.uk

### CODE REVIEW

**Comment by shamser** [comment]
@ghalliday minor change.. please can you review.

File: esp/src/Visualization [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit a40abcd1b85efe2802f5c5cba1c3fa529ce01af3
```
Comment (richardkchapman) [inline esp/src/Visualization]: This change should not be in this commit

**Comment by ghalliday** [comment]
@shamser can you add a test case to ecl/regress (if there isn't one already) that triggers this error in a valid situation, to ensure it is still handled correctly.

**Comment by shamser** [comment]
@ghalliday regression program created (and accidental commit of Visualization sub-module removed.)

**Comment by ghalliday** [comment]
@shamser the example in ecl/regress should be an example which is incorrect.  E.g.,

ds2 := nofold(dataset(100, transform(recordof(sqSimplePersonBookDs), SELF := [])));
fetched := fetchit( ds2, 'Halliday' );

Which generates a horrible internal error when you try it.  The check still needs to happen, but it needs to happen later.

**Comment by shamser** [comment]
@ghalliday Check added.  Please can you review.

**Pull request merged**